Clover/app/src/main/java/org/floens/chan/core/site/ImageSearch.java

```diff
@@ -25,6 +25,8 @@
 
     public abstract int getId();
 
+    public abstract String getType();
```

Comment:
Make this an enum or a static final int.
You can put the enum on the ImageSearch class.